### DISCUSSION TITLE
User log outs after access token is expired and new access token is not generated

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16737] [Archive]
An access token is generated when a user signs in and its time period is of 15 mins.

Normally, when access token expires, refresh token is used to generate new access token by using (/auth/refresh) REST API Endpoint where refresh token time period is 7 days.

But in my case, when access token expires, user logouts exactly after 15 mins.

I want my user to stay logged in there for 7 days (i.e. refresh_token time period) and don't want  to logout after 15 mins.

What am i missing?
If not, then any possible suggestions will be appreciated.

**Comment by br41nslug** [comment]
Are you on the latest version? this should have been fixed with this issue https://github.com/directus/directus/issues/16188

**Reply by bhagya1101** [reply]
I am using directus version 9.21.0

**Reply by br41nslug** [reply]
In that case there is probably something preventing the refresh_token cookie from being set. How are you hosting Directus?

**Reply by bhagya1101** [reply]
I am currently running Directus on my localhost. 

Are there any extra steps that we have to code manually for generating new access token using refresh token.
If there, any help will be appreciated.
